golem-worker-service-base/src/gateway_execution/gateway_http_input_executor.rs

```diff
@@ -87,224 +86,319 @@ impl<Namespace: Clone> DefaultGatewayInputExecutor<Namespace> {
         }
     }
 
-    pub async fn execute(
+    async fn handle_worker_binding(
         &self,
-        http_request_details: &HttpRequestDetails,
-        binding: ResolvedBinding<Namespace>,
-    ) -> poem::Response {
-        let middleware_opt = &http_request_details.http_middlewares;
-        let mut request_details = http_request_details.clone();
-
-        match &binding {
-            ResolvedBinding::Static(StaticBinding::HttpCorsPreflight(cors_preflight)) => {
-                cors_preflight
-                    .clone()
-                    .to_response(http_request_details, &self.gateway_session_store)
-                    .await
-            }
+        namespace: &Namespace,
+        request: &mut RichRequest,
+        binding: &WorkerBindingCompiled,
+    ) -> GatewayHttpResult<RibResult> {
+        let mut rib_input: serde_json::Map<String, Value> = serde_json::Map::new();
 
-            ResolvedBinding::Static(StaticBinding::HttpAuthCallBack(auth_call_back)) => {
-                self.handle_http_auth_call_binding(
-                    &auth_call_back.security_scheme_with_metadata,
-                    http_request_details,
-                )
+        // phase 1. we only have the request details available
```

Comment:
The rib interpreting is now explicitly done in multiple phases (this was already done before, just distributed over many places).
As more data becomes available we expand the context that the script has access to. I don't think we are documenting this currently, so I'll try to do that as part of my doc tickets